From 9bac1a86d866a47697308a92aaabeab385e31a0c Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Wed, 18 Dec 2019 14:28:12 +0200
Subject: [PATCH] Revert "[SW-0] gaudi: pci_pll: skip WA for metal fix asic"

Since we remove the PLL W/A at all (no more old SoCs - only MF), we no
longer need to take special care about the W/A and PCIe PLL init.
Remove the code.

This reverts commit e6c5fca615e1ecc64dff979f2d67222892405b29.

Change-Id: I73cddf1ba7951cbbb82b0512e0d0f702c549d2ab
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/gaudi/asic.c b/gaudi/asic.c
index 3936329..49ff186 100644
--- a/gaudi/asic.c
+++ b/gaudi/asic.c
@@ -269,19 +269,10 @@
 	return regmap_readl(mmPSOC_GLOBAL_CONF_COLD_RST_FLOPS_2) == 1;
 }
 
-static bool gaudi_pcie_strap_is_unset(void)
-{
-	u32 boot_strap = regmap_readl(mmPSOC_GLOBAL_CONF_BOOT_STRAP_PINS);
-
-	/* Check if already done by BTL */
-	return !(boot_strap & 0x2);
-}
-
 bool hl_pll_is_configurable(struct hl_pll_cfg *cfg)
 {
 	switch (cfg->base) {
 	case mmPSOC_PCI_PLL_BASE:
-		return gaudi_pcie_strap_is_unset();
 	case mmPSOC_CPU_PLL_BASE:
 		return false;
 	case mmNIC2_HBM_PLL_BASE:
@@ -295,30 +286,11 @@
 	return true;
 }
 
-static void gaudi_pci_pll_nowa(void)
-{
-	int i, j;
-
-	if (!gaudi_pcie_strap_is_unset())
-		return;
-
-	for (i = 0; i < ARRAY_SIZE(gaudi_pll_cfg); i++) {
-		if (gaudi_pll_cfg[i].base == mmPSOC_PCI_PLL_BASE) {
-			for (j = 0; j < HL_PLL_DIV_SIZE; j++)
-				gaudi_pll_cfg[i].div_wa[j] = HL_PLL_WA_NOWA;
-
-			break;
-		}
-	}
-}
-
 int hl_pll_get_cfg(struct hl_pll_cfg **hl_pll_cfg)
 {
 	int pll_num = ARRAY_SIZE(gaudi_pll_cfg);
 
 	*hl_pll_cfg = gaudi_pll_cfg;
-
-	gaudi_pci_pll_nowa();
 
 	return pll_num;
 }
